station/clients/airflow/docker_trains.py

```diff
@@ -3,47 +3,127 @@
-        # Default config specifying only the identifier of the the train image and using the latest tag
+        # Default config specifies only the identifier of the the train image and uses the latest tag
         config = {
-            "repository": f"{os.getenv('STATION_ID')}/{train_id}",
+            "repository": f"{os.getenv('HARBOR_BASE_URL')}/station_{os.getenv('STATION_ID')}/{train_id}",
             "tag": "latest"
         }
 
-    #  Execute the train using the airflow rest api
-    run_id = airflow_client.trigger_dag("run_train", config=config)
+    if config["repository"] is None or config["tag"] is None:
+        raise HTTPException(status_code=400, detail="Train run parameters are missing.")
+
+    return config
+
+
+def update_train(db, db_train, run_id):
+    """
+    Update train parameters
+    :param db: database session
+    :param db_train: db_train object to update
+    :param run_id: run_id of the triggered run
+    :return:
+    """
+    db_train.is_active = True
+    run_time = datetime.now()
+    db_train.updated_at = run_time
 
     # Update the train state
+    train_state = update_state(db, db_train, run_time)
+
+    # Create an execution
+    execution = dte_model(train_id=db_train.id, airflow_dag_run=run_id)
+    db.add(execution)
+    db.commit()
+    db.refresh(execution)
+
+    db.commit()
+
+    return db_train
+
+def run_train(db: Session, train_id: Any, execution_params: DockerTrainExecution) -> DockerTrainSavedExecution:
+    """
+    Execute a PHT 1.0 docker train using a configured airflow instance
+
+    :param db: database session
+    :param train_id: identifier of the train
+    :param execution_params: given config_id or config_json can be used for running train
+    :return:
+    """
+    # Use default config if there is no config defined.
+    if execution_params is None:
+        execution_params = DockerTrainExecution(config_id="default")
+        print("No config defined. Default config is used.")
+
+    config = validate_run_config(db, train_id, execution_params)
+
+    # Extract the train from the database
     db_train = docker_trains.get_by_train_id(db, train_id)
     if not db_train:
         raise HTTPException(status_code=404, detail=f"Train with id '{train_id}' not found.")
-    db_train.is_active = True
-    db_train.updated_at = datetime.now()
-    if db_train.state:
-        db_train.state.run_id = run_id
-    db.commit()
 
-    return run_id
+    # Execute the train using the airflow rest api
+    try:
+        run_id = airflow_client.trigger_dag("run_pht_train", config=config)
+        db_train = update_train(db, db_train, run_id)
+    except:
+        raise HTTPException(status_code=503, detail="No connection to the airflow client could be established.")
+
+    try:
+        last_execution = db_train.executions[-1]
+    except:
+        last_execution = None
+        print("No executions registered.")
+
+    return last_execution
```

Comment:
Should this not be an error? Does the train need to be refreshed since your adding an execution previously?